scripts/gen-service-kustomize.sh

```diff
@@ -71,6 +72,23 @@ cat <<EOF >>kustomization.yaml
 EOF
 fi
 
+if [ "$NAME" = "galera" ] && [ -n "${DBREPLICAS:-}" ]; then
+cat <<EOF >>kustomization.yaml
+    - op: replace
+      path: /spec/replicas
+      value: ${DBREPLICAS}
+EOF
+fi
+
+if [ "$NAME" = "openstackcontrolplane" ] && [ -n "${DBREPLICAS:-}" ]; then
+cat <<EOF >>kustomization.yaml
+    - op: replace
+      # only update the main galera DB for the time being
+      path: /spec/galera/templates/openstack/replicas
```

Comment:
This will cause a `kustomize` error if for some reason `DBREPLICAS` is set and yet all or part of the `path` is missing in the source `OpenStackControlPlane` YAML.  Though I suppose that would mostly be user error since they would have to be explicitly providing the former and excluding the latter.

Reply:
Right... I don't see a fast and simple means to ensure this path is always present in the resource that is getting built by 'oc kustomize' without running it in gen-service-kustomize.
What about hardening the check by explicitly requiring DBSERVICE=galera and live with that until we deprecate MariaDB in favour of Galera?

Reply:
would it be better to just have an alternate config/sample in openstack-operator that configures Galara instead? Also, I think as much as possible having the default we set via openstack-operator (code can live in mariadb operator and get applied at creation time) would be ideal

Reply:
So you mean no tweak of DBREPLICAS, and instead a single config for 3-node galera? And eventually when we deprecate mariadb in favor of galera we update all mariadb config files to use 1-node galera instead?

I'm good with that, I just like to have a simple means to deploy the openstack-operator with galera, so the DBSERVICE env var seems reasonable until we deprecate mariadb.

Reply:
> would it be better to just have an alternate config/sample in openstack-operator that configures Galara instead? Also, I think as much as possible having the default we set via openstack-operator (code can live in mariadb operator and get applied at creation time) would be ideal

I just updated the PR select different config samples for 1-node galera (default) or 3-node galera. It depends on https://github.com/openstack-k8s-operators/openstack-operator/pull/279.

@dprince is that what you had in mind?